tests/test.py

```diff
@@ -726,861 +754,102 @@ def test_marathon_objects(
-                          self.test_virtual[0]['partition'])
-        self.assertEquals(self.test_partition,
-                          self.test_virtual[1]['partition'])
-        self.assertEquals(expected_name1, self.test_pool[0]['name'])
-        self.assertEquals(expected_name2, self.test_pool[1]['name'])
-        self.assertEquals(self.test_partition, self.test_pool[0]['partition'])
-        self.assertEquals(self.test_partition, self.test_pool[1]['partition'])
-        self.assertEquals(expected_name1, self.test_monitor[0]['name'])
-        self.assertEquals(expected_name2, self.test_monitor[1]['name'])
-        self.assertEquals(self.test_partition,
-                          self.test_monitor[0]['partition'])
-        self.assertEquals(self.test_partition,
-                          self.test_monitor[1]['partition'])
+    def test_marathon_configs(self):
+        """Test: Verify expected BIG-IP config created from Marathon state."""
+        # Verify configuration
+        for key, value in marathon_test_data.items():
+            self.verify_marathon_config(key, value)
 
     def start_two_apps_with_two_partitions(
-            self, partitions, expected_name1, expected_name2,
-            cloud_state='tests/marathon_two_apps_two_partitions.json',
-            bigip_state='tests/bigip_test_blank.json',
-            hm_state='tests/bigip_test_blank.json'):
+            self, partitions, expected_name1, expected_name1_count,
+            expected_name2, expected_name2_count,
+            cloud_state='tests/marathon_two_apps_two_partitions.json'):
         """Test: Start two Marathon apps on two partitions."""
-        # Setup two partitions
-        self.bigip._partitions = partitions
-
         # Get the test data
-        self.read_test_vectors(cloud_state, bigip_state, hm_state)
-
-        # Do the BIG-IP configuration
+        self.read_test_vectors(cloud_state)
         apps = ctlr.get_apps(self.cloud_data, True)
-        cfg = ctlr.create_config_marathon(self.bigip, apps)
-        self.bigip.regenerate_config_f5(cfg)
-
         self.check_labels(self.cloud_data, apps)
 
+        name1_count = 0
+        name2_count = 0
+
+        # Create CCCL instance for each partition and count the number of
+        # times the virtual-server name is found
+        for partition in partitions:
+            with patch.object(ManagementRoot, '_get_tmos_version'):
+                bigip = ManagementRoot(
+                    '1.2.3.4',
+                    'admin',
+                    'default',
+                    port=443,
+                    token='tmos')
+                cccl = F5CloudServiceManager(
+                    bigip,
+                    partition,
+                    prefix='',
+                    schema_path=schema_path)
+
+                cfg = ctlr.create_config_marathon(cccl, apps)
+                if len(cfg['virtualServers']) > 0:
+                    if expected_name1 == cfg['virtualServers'][0]['name']:
```

Comment:
It would be better if we double-checked that the CCCL we expect to handle the app is the one that did.

(For instance, if CCCL 1 created both apps, and CCCL2 created neither, this test would pass but it shouldn't)

Minor.